### PR TITLE
asterisk: 16.26.1 -> 16.29.0, 18.12.1 -> 18.15.0, 19.4.1 -> 19.7.0, init 20.0.0

### PR DESCRIPTION
**Pull request by yorickvP**
###### Description of changes

Ran update script. Not changing lts,stable,default versions for 22.11 but they should change for 23.05.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/asterisk/default.nix [hunk 1]
```diff
@@ -117,6 +122,7 @@ let
     inherit version sha256;
     externals = {
       "externals_cache/pjproject-2.12.tar.bz2" = pjproject_2_12;
+      "externals_cache/pjproject-2.12.1.tar.bz2" = pjproject_2_12_1;
```
Comment (risicle) [inline pkgs/servers/asterisk/default.nix:125]: Both pjsip 2.12 and 2.12.1 have significant CVEs, any idea if asterisk upstream have addressed them?

**Comment by yorickvP** [comment]
@ofborg build asterisk_16 asterisk_18 asterisk_19 asterisk_20

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, nixos x86_64. Patches seem to be applied successfully.

Would be nice to have a nixos test for this and perhaps de-vendor the pjsip to some degree.

**Comment by yorickvP** [comment]
Test got stuck in https://github.com/NixOS/nixpkgs/pull/171234

**Pull request merged**